drivers/regulator/regulator_fixed.c

```diff
@@ -31,6 +31,10 @@ static int regulator_fixed_enable(const struct device *dev)
 	const struct regulator_fixed_config *cfg = dev->config;
 	int ret;
 
+	if (!cfg->enable.port) {
+		return -ENOTSUP;
+	}
+
```

Comment:
what is the point of a fixed regulator that can't be controlled? Also, this seems to deviate from Linux where either clocks/gpio/power domain are required.

Reply:
A driver can request the current voltage of regulator, and for such drivers, we can utilize distinct regulators for various nodes, for example, fixed w/o control for one node of the driver and gpio regulator for another node

Many fixed regulators do not require clocks, GPIO, or power domains within the Linux kernel. For instance, refer to: https://elixir.bootlin.com/linux/latest/source/arch/arm64/boot/dts/renesas/salvator-common.dtsi#L181

Reply:
ok, fair enough. Doxygen should also be updated to mention this is a valid case:

```
@retval -ENOTSUP If regulator enablement can not be controlled
```

Reply:
> Doxygen should also be updated to mention this is a valid case

is it really needed? as far as I see, we have doxygen style comment only for common enable function, `regulator_enable`:
```
 * @retval 0 If regulator has been successfully enabled.
 * @retval -errno Negative errno in case of failure.
 */
int regulator_enable(const struct device *dev);
```

For some drivers, this could have another meaning. Furthermore, this function (`regulator_fixed_enable`) won't be called in cases where we have the `always-on` property in the device tree (`dts`), see `regulator_enable`:
```
	/* regulator must stay always on */
	if  ((config->flags & REGULATOR_ALWAYS_ON) != 0U) {
		return 0;
	}
```

I believe that `always-on` and `enable-gpios` should indeed be mutually exclusive for the fixed regulator driver, and it would be beneficial to add a build assert:
```
BUILD_ASSERT(DT_PROP_HAS_IDX(DT_DRV_INST(inst), enable_gpios, 0) ^       \
             DT_PROP_OR(DT_DRV_INST(inst), regulator_always_on, 0),      \
             "always-on and enable-gpios should be mutually exclusive"); \
```

What are your thoughts on this?

Reply:
> > Doxygen should also be updated to mention this is a valid case
> 
> is it really needed? as far as I see, we have doxygen style comment only for common enable function, `regulator_enable`:
> 
> ```
>  * @retval 0 If regulator has been successfully enabled.
>  * @retval -errno Negative errno in case of failure.
>  */
> int regulator_enable(const struct device *dev);
> ```

`-errno` case is really reserved for propagated errors (e.g. from a failed transfer). The `-ENOTSUP` case here is not strictly a failure.

> 
> For some drivers, this could have another meaning. Furthermore, this function (`regulator_fixed_enable`) won't be called in cases where we have the `always-on` property in the device tree (`dts`), see `regulator_enable`:
> 
> ```
> 	/* regulator must stay always on */
> 	if  ((config->flags & REGULATOR_ALWAYS_ON) != 0U) {
> 		return 0;
> 	}
> ```
> 

This case is differentiated because it returns 0. Actually, always-on/boot-on don't make much sense in a fixed regulator without gpios.

> I believe that `always-on` and `enable-gpios` should indeed be mutually exclusive for the fixed regulator driver, and it would be beneficial to add a build assert:
> 
> ```
> BUILD_ASSERT(DT_PROP_HAS_IDX(DT_DRV_INST(inst), enable_gpios, 0) ^       \
>              DT_PROP_OR(DT_DRV_INST(inst), regulator_always_on, 0),      \
>              "always-on and enable-gpios should be mutually exclusive"); \
> ```
> 
> What are your thoughts on this?

I think it makes sense, but right now, it's used to turn on the regulator during init time (set to active forever). Also, I think we're using the `regulator-boot-on` incorrectly, as it should be used to flag that the regulator is already on (from a previous stage).

Reply:
> it's used to turn on the regulator during init time (set to active forever). 

I see, than there is no sense to add a build assert,
I have updated `regulator_enable` description